DVCSMP-2874 Aeon multisensor 6 missing method exception

greens opened the pull request:
The `secure` method was never defined in this DTH and also the device is only conditionally a listening device, so I've reflected that in the `poll` method.

greens committed:
DVCSMP-690 Aeon multisensor improvements

Bundles configuration and wakeup commands into one multiAction

dkirker reviewed (commented):
@jackchi Does the fixed ping() command look ok? If it is running off battery there is nothing we can do until it sends a wakeup notification.

However, one thought:
Would device health accept this as an "it is online" event?

createEvent(descriptionText: "${device.displayName} woke up", isStateChange: false)

If yes (would it even get it because isStateChange is false?), then I think this looks good. If no, then we should probably add a state variable that is a flag indicating that we should request battery state in the next wakeup.

@jackchi @mckeed Thoughts?

mckeed commented:
Wakeup events do work for device health, but this device sends multisensor events every 8 minutes, so those are what we're relying on to keep the device online.

It doesn't make a lot of sense to send a batteryGet as the ping when the batteries aren't being used. Maybe requesting the temperature would be better, since we shouldn't get to the point of pinging the device unless the multisensor events aren't getting received.

greens commented:
That's a good point. I'll make some changes to use temperature instead.

greens committed:
DVCSMP-2874 Aeon Multisensor 6 NoSuchMethod Exception

Changes the checkInterval to match the user-set report interval.

Changes ping to use temperature as ping will only work when the device is on ac power, so a battery poll would be not as useful.

greens commented on devicetypes/smartthings/aeon-multisensor-6.src/aeon-multisensor-6.groovy:
```diff
@@ -142,10 +142,15 @@ def installed(){
+	checkInterval = 2 * 60 * 60 + 2 * 60
+	if (reportInterval) {
+		checkInterval = 2*(timeOptionValueMap[reportInterval]/60) + (2 * 60)
+	}
```
check my logic here, esp. re: accessing preference values.

mckeed commented on devicetypes/smartthings/aeon-multisensor-6.src/aeon-multisensor-6.groovy:
```diff
@@ -142,10 +142,15 @@ def installed(){
 	log.debug "Updated with settings: ${settings}"
 	log.debug "${device.displayName} is now ${device.latestValue("powerSource")}"
 
+	checkInterval = 2 * 60 * 60 + 2 * 60
```
missing `def`

mckeed commented:
I tried this out and ran into an issue: when the device is battery powered the check interval is changed right away but the reporting interval isn't changed until the next wakeup, which makes the device go offline when switching to a shorter interval

greens commented:
Ahh, okay. Missed that. So we should update the checkInterval inside `configure`.

greens committed:
DVCSMP-2874 Aeon multisensor 6 ping fixes

Changes the checkinterval update to happen inside configure in cases where the device is in wakeup mode so that we don't change the check interval without changing the report interval.

tpmanley commented on devicetypes/smartthings/aeon-multisensor-6.src/aeon-multisensor-6.groovy:
```diff
@@ -399,6 +402,14 @@ def configure() {
+	// we do this here in case the device is in wakeup mode
+	def checkInterval = 2 * 60 * 60 + 2 * 60
+	if (reportInterval) {
+		checkInterval = 2*(timeOptionValueMap[reportInterval]/60) + (2 * 60)
```
The value for `"18 hours"` in `getTimeOptionValueMap` is actually only 6 hours

greens replied:
That's a good catch. Looks like it's wrong in the aeon multisensor (original) code too. I'll go ahead and fix both under this commit.

mckeed replied:
Math is wrong here. You're dividing the seconds by 60 but checkInterval is in seconds too.

greens committed:
DVCSMP-2874 Aeon multisensor 6 missing method exception

Fixes a typo in the time lookup table.

greens committed:
DVCSMP-2874 Aeon multisensor 6 missing method exception

Fixes some bad math

greens committed:
Merge branch 'master' into DVCSMP-2874

greens commented:
Can I get a second look at this?

greens commented:
@mckeed Incorporated your suggestions.
